### PR TITLE
Feature/lineui enhancement

### PR DESCRIPTION
**Pull request by fti-pfarne**
### Description

- This PR is enhancement in Line ui .

- During selecting a line UI by mouse click, it was difficult to select line by clicking exactly on line. For overcoming this issues added transparent border around line.which will show on hover of line that help to click on line easily. 

- For selecting line, I have added on click functionality for arrows of line as well as name of line.

- Related Bugherd Requirement links
[Requirement1](https://www.bugherd.com/projects/289019/tasks/159)
[Requirement2](https://www.bugherd.com/projects/289019/tasks/257)

This changes are required on Traffic Counter Edge application.

### Considerations on Implementation

- Added a prop lineClickSensingBorder to give width to transparent border and default width of border is 65px.

### ScreenShot
Screenshot from 2022-11-28 12-00-54

Screenshot from 2022-12-08 21-21-48

Screenshot from 2022-12-09 12-21-38

Screenshot from 2022-12-09 12-19-58

### CODE REVIEW

File: src/LineUI/LineUnit.tsx [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
   stroke: ${({theme, styling}) => theme.custom.lines[styling].contrastLine.stroke};
   mix-blend-mode: multiply;
+  stroke-width: ${({ lineClickSensing }) => lineClickSensing}px;
+  stroke-opacity: ${({ showLineBorder }) => showLineBorder ? '0.35' : '0'};
 `;
 
 const HighlightLine = styled.line<{styling: string}>`
   pointer-events: none;
-  stroke: ${({theme, styling}) => theme.custom.lines[styling].highlightLine.stroke};
+  stroke: #3781b5;
```
[the remark this answers is not in the document]
Comment (fti-pfarne) [reply src/LineUI/LineUnit.tsx]: Done

File: src/themes/light/custom.js [hunk 2]
```diff
@@ -10,6 +10,9 @@ export const custom = {
       highlightLine: {
           stroke: 'hsla(205deg, 45%, 90%, 90%);'
       },
+      highlightLineBorder: {
+          stroke:  'hsla(204.762deg, 53%, 46%, 1);'
+      },
```
Comment (fti-bcarpenter) [inline src/themes/light/custom.js:15]: You have added this for only primary line. I think this should also be added for secondary and danger line.
Also for different style you might have to use different stroke color.
Have you checked your changes with different line styles?

File: src/LineUI/LineUI.tsx [hunk 3]
```diff
@@ -87,7 +91,7 @@ const LineUI: React.FC<LineUIProps> = ({
     showPoint = false,
     fixedImgDimensions,
     boundaryOffset = 0,
-    showDirectionMark = false
+    showDirectionMark = false,
```
Comment (fti-bcarpenter) [inline src/LineUI/LineUI.tsx:94]: There is no need of , at end here

**Review by fti-bcarpenter — CHANGES_REQUESTED** [review]

File: src/LineUI/LineUI.tsx (hunk 3 above)
Comment (fti-pfarne) [reply src/LineUI/LineUI.tsx]: Done

File: src/themes/light/custom.js (hunk 2 above)
Comment (fti-pfarne) [reply src/themes/light/custom.js:15]: Done

File: src/themes/light/custom.js [hunk 4]
```diff
@@ -106,6 +109,9 @@ export const custom = {
       highlightLine: {
           stroke: 'hsla(0, 45%, 90%, 90%);'
       },
+      highlightLineBorder: {
+        stroke:  'hsla(204.762deg, 53%, 46%, 1);'
+      },
```
Comment (fti-bcarpenter) [inline src/themes/light/custom.js:114]: You should not use same color for all 3 style. Have you verified it. 
Please add the snapshot of 3 different style of Lines.

**Review by fti-bcarpenter — CHANGES_REQUESTED** [review]

File: src/themes/light/custom.js (hunk 4 above)
Comment (fti-pfarne) [reply src/themes/light/custom.js:105]: Yes sir. Done

**Review by fti-bcarpenter — APPROVED** [review]

**Review by atomworks — CHANGES_REQUESTED** [review]
This works really nicely. Thank you!

Just a couple of usability requests...
* When a line is selected, can we disable the hover area? It's a bit strange when it's already active although not the end of the world. 
* When we hover and it shows that select area, can we also have the mouse cursor change to pointer to show it can be selected?

If we do not have time for one or both of those before this is needed for production work, we can follow this up later although it would be great to have done. If there isn't time, just re-request a review and let me know.

This component is a challenging one to work with and seeing what you've done here makes me happy. Thank you.

**Review by atomworks — APPROVED** [review]
As per our conversation, let's follow up the final point in a future PR. 
> When a line is selected, can we disable the hover area? It's a bit strange when it's already active although not the end of the world.

**Pull request merged**